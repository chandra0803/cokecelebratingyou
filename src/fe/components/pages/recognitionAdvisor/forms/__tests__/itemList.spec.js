/*global jest*/

jest.mock( 'react-redux' );
import React from 'react';
import ItemList from '../itemList.js';
import renderer from 'react-test-renderer';

window.recognitionAdvisor = {
  raUrl: 'ra/reminders.action',
  raTilePageDisplay: 'yes',
  raDetailPageDisplay: 'no',
  raEndModelPageDisplay: 'no',
  content: []
};

it( 'renders the dom like the snapshot when the props have default values.', () => {
  const item = {
    'id': 92085,
    'firstName': 'Erica',
    'lastName': 'Jalecia Edwards',
    'participantGroupType': 2,
    'avatarUrl': null,
    'positionType': 'Director',
    'otherMgrApprovedDate': null,
    'currentMgrApprovedDate': '01/08/2018',
    'otherMgrLastRecogDays': null,
    'currentMgrLastRecogDays': 225,
    'otherMgrAwardPoints': null,
    'currentMgrAwardPoints': 10,
    'claimIdByCurrentMgr': '10354',
    'claimIdByOtherMgr': null,
    'noOfDaysPastDueDate': null,
    'participantRecDaysColour': 'red',
    'participantRecDaysColourForOthers': null,
    'claimUrlByCurrentMgr': 'https://g5betaqa.performnet.com/g5beta/login.do',
    'claimUrlByOtherMgr': null,
    'countryId': 6037,
    'nodeId': [ '7202' ],
    'recApprovalStatusType': null
  };

  const component = renderer.create(
    <div>
      <ItemList item={ item } />
    </div>
  );

  const tree = component.toJSON();
  expect( tree ).toMatchSnapshot();
} );
