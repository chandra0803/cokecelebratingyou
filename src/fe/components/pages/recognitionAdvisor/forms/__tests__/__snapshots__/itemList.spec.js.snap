// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the dom like the snapshot when the props have default values. 1`] = `
<div>
  <div>
    <a
      className="arrow icon-arrow-1-circle-right"
      data-id={92085}
      onClick={[Function]}
    />
    <div
      className="avatarwrap"
    >
      <span
        className="avatar-initials"
      >
        <span>
          E
           
          J
        </span>
      </span>
    </div>
    <a
      className="itemName leaderName profile-popover"
      data-participant-ids="[ 92085 ]"
    >
      Erica
       
      Jalecia Edwards
    </a>
    <span
      className="desgination"
    >
      Director
    </span>
  </div>
</div>
`;
